### PR TITLE
Add ability to return multiple values

### LINKED ISSUE #50: Add ability to return multiple values
It would be nice to be able to return multiple values. 

Consider this toy example of an rle function created as mostly as a quickr function. 

```r
rle_combined <- function(x) {
  declare(
    type(x = integer(n)),
    type(lengths = integer(n)),
    type(values = integer(n))
  )
  
  n <- length(x)
  
  lengths <- 0L
  placeholder_value <- max(x) + 1L # placeholder
  values <- placeholder_value
  
  run_idx <- 1L
  run_length <- 1L
  values[1L] <- x[1L]  # First run value
  
  for (i in 2:n) {
    if (x[i] == x[i-1]) {
      run_length <- run_length + 1L
    } else {
      lengths[run_idx] <- run_length
      run_idx <- run_idx + 1L
      run_length <- 1L
      values[run_idx] <- x[i]
    }
  }
  # last run
  lengths[run_idx] <- run_length
  
  list(lengths = lengths, values = values, placeholder = placeholder_value)
}
```

Then we can use the returned values to recreate `base::rle`

```r
q_rle_combined <- quick(rle_combined)

my_rle<- function(x){
  combined_list <- q_rle_combined(x)

  out <- list(
    lengths = combined_list$lengths[combined_list$lengths > 0],
    values = combined_list$values[combined_list$values != combined_list$placeholder]
  )
  
  class(out) <- "rle"
  out
}
```

### PR DESCRIPTION
**Pull request by mns-nordicals**
This pull request adds the ability to return multiple values from a quickr function. 

I'll admit that it was mostly written by AI, but in my testing it seem to work well. It clears all test and I have added a new test file with a few tests. 

This closes #50

### CODE REVIEW

File: R/c-wrapper.R [hunk 1]
```diff
@@ -64,10 +67,27 @@ make_c_bridge <- function(fsub, strict = TRUE, headers = TRUE) {
...
+  } else {
+    return_var_names_un <- unname(return_var_names)
+    append(c_body) <- c(
+      glue("SEXP _ans = PROTECT(Rf_allocVector(VECSXP, {length(return_var_names_un)}));"),
+      imap(return_var_names_un, function(nm, i) {
+        glue("SET_VECTOR_ELT(_ans, {i-1}, {nm});")
+      }),
+      glue("SEXP _names = PROTECT(Rf_allocVector(STRSXP, {length(return_var_names_un)}));"),
+      imap(return_var_names_un, function(nm, i) {
+        glue("SET_STRING_ELT(_names, {i-1}, Rf_mkChar(\"{nm}\"));")
```
Comment (t-kalinowski) [inline R/c-wrapper.R:84]:
```suggestion
        glue('SET_STRING_ELT(_names, {i-1}, Rf_mkChar("{nm}"));')
```

Comment (t-kalinowski) [reply R/c-wrapper.R:84]: It seems this approach requires that the list element name is the same as the symbol. i.e., the last expression is like `list(foo = foo, bar = bar)`. 

Is that correct? What do you think about supporting different names for the list elements? E.g., `list(foo = abc, bar = xyz)`.

File: tests/testthat/test-multiple-return.R [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  ## List is being accessed
+  fn <- function(x) {
+    declare(
+      type(x = double(n)),
+      type(y = double(n)),
+      type(z = double(n))
+    )
+    y <- x + 1
+    z <- x + 2
+    out <- list(y = y, z = z)
```
Comment (t-kalinowski) [inline tests/testthat/test-multiple-return.R:72]: Can you please add tests to check behavior when:

- return value is an unnamed list
- return value is a list with names that don't match the symbol name.

**Review by t-kalinowski — COMMENTED** [review]
Very nice work! I think this is almost ready to merge. Just had some minor questions about how we handle named lists.

**Comment by t-kalinowski** [comment]
@codex review

**Comment by t-kalinowski** [comment]
@codex add tests for the single list edge case, show the test failing (by pushing), then implement the fix (in another commit). 
For the non-syntatic names, just throw an error message that explains only syntatic names are valid.

**Comment by t-kalinowski** [comment]
@codex review again
